lib/ansible/module_utils/eos.py

```diff
@@ -180,6 +191,9 @@ def _load_params(self):
             if key in NET_COMMON_ARGS.keys():
                 if not params.get(key) and value is not None:
                     params[key] = value
+        for key, env_var in NET_ENV_ARGS.items():
+            if not params.get(key) and os.environ.get(env_var):
+                params[key] = os.environ[env_var]
```

Comment:
One more thing needs to go in this section:  defaults and required.
* Since we have to set the argspec to default to a sentinel value (None) so that we know the difference between "user didn't specify anything" and "user specified something that's falsey" we also need to add any default values here.  Judging from the previous argspec, this is just if authorize is still None then set authorize equal to False.
* A similar thing happens with required items.  We have to set the argspec to required=False so that we can also look for the values in these envvars.  Once we've looked added any envvars to the parameters we have to manually check that any of those that we require are now set.  Judging from the argspec changes, we just need to check if username is still None then call module.fail_json() to let the user know that a username has to be set.